app/src/main/java/org/fossasia/pslab/communication/sensors/ADS1115.java

```diff
@@ -0,0 +1,243 @@
+        gain = Gain;
+    }
+    public void setChannel(String Channel){
+        /*options 'UNI_0','UNI_1','UNI_2','UNI_3','DIFF_01','DIFF_23'*/
+        channel = Channel;
+    }
+
+    public void setDataRate(int Rate){
+        /*data rate options 8,16,32,64,128,250,475,860 SPS*/
+        rate = Rate;
+    }
+
+    public double readADC_SingleEnded(int chan) throws IOException, InterruptedException {
+        if (chan>3) {
+            return -1;
+        }
+        //start with default values
+        int config = REG_CONFIG_CQUE_NONE             //Disable the comparator (default val)
+                | REG_CONFIG_CLAT_NONLAT              //Non-latching (default val)
+                | REG_CONFIG_CPOL_ACTVLOW             //Alert/Rdy active low   (default val)
+                | REG_CONFIG_CMODE_TRAD               // Traditional comparator (default val)
+                | REG_CONFIG_MODE_SINGLE              // Single-shot mode (default)
+                | sdrSelection.get(rate) ;            //1600 samples per second (default)
+
+        //Set PGA/voltage range
+        config = config | gains.get(gain);
+
+        if(chan == 0)
+            config = config | REG_CONFIG_MUX_SINGLE_0;
+        else if (chan == 1)
+            config = config | REG_CONFIG_MUX_SINGLE_1;
+        else if (chan == 2)
+            config = config | REG_CONFIG_MUX_SINGLE_2;
+        else if (chan == 3)
+            config = config | REG_CONFIG_MUX_SINGLE_3;
+
+        //Set 'start single-conversion' bit
+        config = config | REG_CONFIG_OS_SINGLE;
+        writeRegister(REG_POINTER_CONFIG, config);
+        TimeUnit.SECONDS.sleep((long) (1./rate + 0.002));       //convert to mS to S
+        return readRegister(REG_POINTER_CONVERT)*gainScaling.get(gain);
+    }
+
+    public short readADC_Differential(String chan) throws IOException, InterruptedException {
+        //start with default values
+        int config = REG_CONFIG_CQUE_NONE              //Disable the comparator (default val)
+                | REG_CONFIG_CLAT_NONLAT               //Non-latching (default val)
+                | REG_CONFIG_CPOL_ACTVLOW              //Alert/Rdy active low   (default val)
+                | REG_CONFIG_CMODE_TRAD                // Traditional comparator (default val)
+                | REG_CONFIG_MODE_SINGLE               // Single-shot mode (default)
+                | sdrSelection.get(rate) ;             //1600 samples per second (default)
+
+        //Set PGA/voltage range
+        config = config | gains.get(gain);
+
+        if(chan.equals("01"))
+            config = config | REG_CONFIG_MUX_DIFF_0_1;
+        else if (chan.equals("23"))
+            config = config | REG_CONFIG_MUX_DIFF_2_3;
+
+        //Set 'start single-conversion' bit
+        config = config | REG_CONFIG_OS_SINGLE;
+        writeRegister(REG_POINTER_CONFIG, config);
+        TimeUnit.SECONDS.sleep((long) (1./rate + 0.002));       //convert to mS to S
```

Comment:
If you parse 1. / rate + 0.002 to long, it will always return 0 taking all possible rate values into consideration as long is an integer datatype.
Use the following approach.
Use TimeUnit.MILLISECOND.sleep((long)((1. / rate + 0.002) * 1000))
If I take rate as 8, the sleep time will be 0.127 seconds or 127 milliseconds.
Change the code using the similar approach in line 200 and 147.